lib/lua/utils.lua

```diff
@@ -136,4 +136,39 @@ function util.table.get_random(tab)
     end
 end
 
-return util
+-- List of programs that should be spawned each start/restart.
+util.program = {}
+util.program.programs = {}
+
+-- Registers the program to spawn at startup and every time it restarts
+-- @param bin The program to run. Can be an absolute path or a command to run.
+-- @param args The arguments (as a string) to pass to the program.
+function util.program.spawn_at_startup(bin, args)
+  assert(type(bin) == 'string', 'Non string given for program')
+  table.insert(util.program.programs, {
+                 bin = bin,
+                 args = args
+  })
+end
+
+-- Spawns the startup programs
+function util.program.spawn_startup_programs()
+  for index, program in ipairs(util.program.programs) do
+    os.execute(program.bin .. " " .. program.args .. " &")
+  end
+end
+
+-- Stops the startup programs. Does not remove them from the global list.
+function util.program.terminate_startup_programs()
+  for index, program in ipairs(util.program.programs) do
+    -- TODO Store PIDs and kill those instead
```

Comment:
Currently the way this is implemented, any program launched from `spawn_at_startup` is killed in this way (`pkill` currently).

When Way Cooler dies, everything else is killed normally (no `pkill` nonsense). For now, I'm going to keep it in its current form.

The main reason I wanted to have a more fine-grained way to fix this was testing Way Cooler while running Way Cooler would kill my host background/bar program as well as the instance I was testing :(. This is a very small issue though, so I should probably just live with it ;)